adapta/storage/distributed_object_store/datastax_astra/astra_client.py

```diff
@@ -0,0 +1,183 @@
+        :param: reconnect_max_delay_ms: Reconnect delay in ms (max value for exp backoff).
+        :param: socket_connection_timeout: Connect timeout for TCP operations.
+        :param: socket_read_timeout: Read timeout for TCP operations (query timeout).
+        """
+        tmp_bundle_file_name = str(uuid.uuid4())
+        os.makedirs(self._tmp_bundle_path, exist_ok=True)
+
+        with open(os.path.join(self._tmp_bundle_path, tmp_bundle_file_name), "wb") as bundle_file:
+            bundle_file.write(base64.b64decode(self._secure_connect_bundle_bytes))
+
+        cloud_config = {"secure_connect_bundle": os.path.join(self._tmp_bundle_path, tmp_bundle_file_name)}
+        auth_provider = PlainTextAuthProvider(self._client_id, self._client_secret)
+
+        profile = ExecutionProfile(
+            retry_policy=RetryPolicy(),
+            consistency_level=ConsistencyLevel.LOCAL_QUORUM,
+            serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
+            request_timeout=self._socket_read_timeout / 1e3,
+            row_factory=dict_factory,
+        )
+
+        # https://docs.datastax.com/en/developer/python-driver/3.28/getting_started/
+        self._session = Cluster(
+            execution_profiles={EXEC_PROFILE_DEFAULT: profile},
+            cloud=cloud_config,
+            auth_provider=auth_provider,
+            reconnection_policy=ExponentialReconnectionPolicy(
+                self._reconnect_base_delay_ms, self._reconnect_base_delay_ms
+            ),
+            compression=True,
+            application_name=self._client_name,
+            application_version=__version__,
+            sockopts=[
+                (IPPROTO_TCP, TCP_NODELAY, 1),
+                (IPPROTO_TCP, TCP_USER_TIMEOUT, self._socket_read_timeout),
+            ],
+        ).connect(self._keyspace)
+
+        set_session(self._session)
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._session.shutdown()
+        self._session = None
+        shutil.rmtree(self._tmp_bundle_path)
+
+    def get_entity(self, table_name: str) -> Dict:
+        """
+        Reads a single row from a table as dictionary
+        https://docs.datastax.com/en/developer/python-driver/3.28/cqlengine/queryset/
+        """
+
+        named_table = NamedTable(self._keyspace, table_name)
+        return named_table.objects[0]
+
+    def get_entities(self, query: str, mapper: Callable[[Dict], TModel]) -> pandas.DataFrame:
+        """
+        Maps query result to a pandas Dataframe using custom mapper
+        """
+        return pandas.DataFrame([mapper(entity) for entity in self._session.execute(query)])
+
+    def get_entities_raw(self, query: str) -> pandas.DataFrame:
+        """
+        Maps query result to a pandas Dataframe
+        """
+        return pandas.DataFrame(self._session.execute(query))
+
+    @staticmethod
+    def model_dataclass(value: Any, primary_keys: List[str]):
```

Comment:
I think it would be better to create decorator instead of a static method.

Reply:
Now that I think of it, we can't use decorator for imported classes. So while we can add it for OSS, it doesn't have any use for us.